packages/app/src/systems/Asset/components/AssetsAmount/AssetsAmount.tsx

```diff
@@ -25,38 +25,50 @@ export const AssetsAmount: AssetsAmountComponent = ({
   balanceErrors,
 }: AssetsAmountProps) => {
   const hasError = !!balanceErrors?.length;
+  const allEmptyAmounts = amounts.every((assetAmount) =>
+    bn(assetAmount.amount).eq(0),
+  );
 
   return (
-    <Box css={styles.card} data-error={hasError}>
-      {(title || hasError) && (
-        <Box.Flex css={styles.header}>
-          {title && (
-            <Text as="h3" css={styles.title}>
-              {title}
-            </Text>
+    <>
+      {(!allEmptyAmounts || hasError) && (
+        <Box css={styles.card} data-error={hasError}>
+          {(title || hasError) && (
+            <Box.Flex css={styles.header}>
+              {title && (
+                <Text as="h3" css={styles.title}>
+                  {title}
+                </Text>
+              )}
+              {hasError && (
+                <Text
+                  css={{
+                    color: '$intentsError10',
+                    fontSize: '$sm',
+                    fontWeight: '$normal',
+                  }}
+                >
+                  (not enough balance)
+                </Text>
+              )}
+            </Box.Flex>
           )}
-          {hasError && (
-            <Text
-              css={{
-                color: '$intentsError10',
-                fontSize: '$sm',
-                fontWeight: '$normal',
-              }}
-            >
-              (not enough balance)
-            </Text>
+          {!allEmptyAmounts && (
+            <Box.Stack gap="$2">
+              {amounts.map(
```

Comment:
this can still show asset with zero value

if has 2 assets and only 1 of them is zero, wallet will show the 2 assets

Reply:
It won't show the 1 zero asset bc of line 60